src/interceptors/ClientRequest/utils/recordRawHeaders.ts

```diff
@@ -105,14 +136,27 @@ export function recordRawFetchHeaders() {
 
   Request = new Proxy(Request, {
     construct(target, args, newTarget) {
-      const request = Reflect.construct(target, args, newTarget)
-
+      /**
+       * @note If the headers init argument of Request
+       * is existing Headers instance, use its raw headers
+       * as the headers init instead.
+       * This is needed because the Headers constructor copies
+       * all normalized headers from the given Headers instance
+       * and uses ".append()" to add it to the new instance.
+       */
       if (
         typeof args[1] === 'object' &&
         args[1].headers != null &&
-        !request.headers[kRawHeaders]
+        args[1].headers instanceof Headers &&
+        Reflect.has(args[1].headers, kRawHeaders)
       ) {
-        request.headers[kRawHeaders] = inferRawHeaders(args[1].headers)
+        args[1].headers = args[1].headers[kRawHeaders]
+      }
+
+      const request = Reflect.construct(target, args, newTarget)
+
+      if (typeof args[1] === 'object' && args[1].headers != null) {
+        defineRawHeaders(request.headers, inferRawHeaders(args[1].headers))
```

Comment:
Latest Node.js v18 and v20 does _not_ copy Headers symbols (previous versions did!). We need to manually copy the `kRawHeaders` symbol (or set it by creating a new Headers instance) when constructing Request/Response instances.